cooltools/lib/common.py

```diff
@@ -532,15 +532,14 @@ def wrapper(*args, **kwargs):
                 mymap = pool.imap
             else:
                 mymap = map
-
-            func(*args, **kwargs, map=mymap)
-
+            try:
+                result = func(*args, **kwargs, map=mymap)
+            finally:
+                if pool != None:
+                    pool.close()
+            return result 
         else:
-            raise TypeError(
-                "nproc must be specified as a keyword argument (e.g. nproc=1, nproc=5...)"
-            )
-
-        if pool != None:
-            pool.close()
-
+            warnings.warn("nproc is not specified, single thread is used")
```

Comment:
I wonder if logging is more appropriate here ? Also maybe it would be worth logging about creating a pool of nproc workers when nproc>1 as well ?